### ISSUE TITLE
General code improvements

### ISSUE THREAD

**Issue by lojack5** [wrye-bash/wrye-bash#634]
This is meant to be a tracker for general code improvements that aren't big enough issues to warrant their own issue, and not small enough to just make the changes directly.  Most of these initial issues are thoughts I had while writing type-stubs and noticing places for improvements.

Tick off items if they've been addressed.  If we decided to do nothing about it, additionally mark it as such by ~~striking it through~~.

- [ ] `bosh.Omod` and `bosh.OmodConfig` both handle reading and writing omod config files.  Consider merging the interfaces into a common one?
- [ ] Contants in `env` can be replaced with enums:
  - [ ] `FO_*` constants with an `IntEnum`.
  - [ ] `FOF_*` constants with an `IntFlags.`.

**Comment by Utumno** [comment]
There is a big discussion on Flags in #480 + a few merges like 87c7a4cc66bddfc4d18511023e4dc41633c86043 - conclusion was that indeed IntFlag can't serve as a drop in replacement

**Comment by lojack5** [comment]
Yeah, the biggest stopper that I ran into while investigating is that `IntFlag`s *are* `int`s, which are an immutable type. Pretty much prevents us from every providing a `flags.isHidden = True` interface.  The best we could get is compound assignment operators, `flags |= flags.isHidden`.  And once py 3.11 comes we're loosing access to the flag enums on instances, so that would have to change to `flags |= FlagClass.isHidden`, and testing for truthyness would have to change to `if FlagClass.isHidden in flags:`.  Overall I like the mutableness of our flags.  My Idea though is we might be able to implement some metaclass magicry to make definitions easier, like:
```python
class _MarkerFlags(Flags):
    visible: bool
    can_travel_to: bool
    show_all_hidden: bool
```
That'd take some work to get right though, messing with metaclasses has all sorts of pitfalls, and doing it in a way so typecheckers see the flag names as bools while also being able to define them the way you want is even weirder.

**Comment by Utumno** [comment]
I have mentioned metaclasses but should be understood as a joke :D code not ready yet.

Re: flags yes - see also remarks in 87c7a4cc66bddfc4d18511023e4dc41633c86043. Bottom line those are better left alone for now.

I will be adding some todos that I run across in 543 (that is anywhere) in this issue that you might be interested into - it is especially around converters on which I had to plunge as I am trying to finalize 543 - there is some pretty complex code in there:

- [ ] __extractOmods vs _extractOmods 1987c08fa978437daf7bf5b775937d58cd73e327 - don't underestimate this one
